Dockerfile

```diff
@@ -1,40 +1,66 @@
-FROM debian:jessie
+FROM python:2-slim
+MAINTAINER Jannis Leidel <jezdez@mozilla.com>
 
 ENV PYTHONUNBUFFERED 1
 ENV PYTHONPATH /app/
 
-RUN set -x \
-    && apt-get update \
-    && apt-get install locales -y \
-    && locale-gen en_US.UTF-8
-
 EXPOSE 8000
 
-RUN adduser --uid 1000 --disabled-password --gecos '' --no-create-home webdev
+# add a non-privileged user for installing and running the application
+# don't use --create-home option to prevent populating with skeleton files
+RUN mkdir /app && \
+    chown 10001:10001 /app && \
+    groupadd --gid 10001 app && \
+    useradd --no-create-home --uid 10001 --gid 10001 --home-dir /app app
 
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-        build-essential python python-dev python-pip libpq-dev \
-        postgresql-client gettext sqlite3 libffi-dev \
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+# install a few essentials and clean apt caches afterwards
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        apt-transport-https build-essential curl git libpq-dev \
+        postgresql-client gettext sqlite3 libffi-dev  && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-WORKDIR /app
+# Install node from NodeSource
+RUN curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - && \
+    echo 'deb https://deb.nodesource.com/node_4.x jessie main' > /etc/apt/sources.list.d/nodesource.list && \
+    echo 'deb-src https://deb.nodesource.com/node_4.x jessie main' >> /etc/apt/sources.list.d/nodesource.list && \
+    apt-get update && apt-get install -y nodejs
+
+# Create static and npm roots
+RUN mkdir -p /opt/npm /opt/static && \
+    chown -R 10001:10001 /opt
 
-# Get pip8
-COPY bin/pipstrap.py bin/pipstrap.py
-RUN ./bin/pipstrap.py
+# Install Python dependencies
+COPY requirements.txt /tmp/
+# Switch to /tmp to install dependencies outside home dir
+WORKDIR /tmp
+RUN pip install --no-cache-dir -r requirements.txt
 
-COPY requirements.txt /tmp/requirements.txt
+# Install frontend dependencies using NPM
+COPY package.json /opt/npm/
 
-RUN set -x \
-    && pip install -U pip setuptools \
-    && pip install -r /tmp/requirements.txt \
-    && find /usr/local -type f -name '*.pyc' -name '*.pyo' -delete \
-    && rm -rf ~/.cache/
+# Switch to /opt/npm to install dependencies outside home dir
+WORKDIR /opt/npm
+RUN npm install && \
+    chown -R 10001:10001 /opt/npm && \
+    npm cache clean
+
+# Switch back to home directory
+WORKDIR /app
 
 COPY . /app
 
-RUN chown webdev.webdev -R .
-USER webdev
+RUN chown -R 10001:10001 /app
+
+USER 10001
+
+RUN python manage.py collectstatic --noinput
```

Comment:
I tried to build the container and I received the following error:

```
Step 20 : RUN python manage.py collectstatic --noinput
 ---> Running in d19bbee75d22
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 185, in fetch_command
    settings.INSTALLED_APPS
  File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 55, in __getattr__
    self._setup(name)
  File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 43, in _setup
    self._wrapped = Settings(settings_module)
  File "/usr/local/lib/python2.7/site-packages/django/conf/__init__.py", line 120, in __init__
    raise ImproperlyConfigured("The SECRET_KEY setting must not be empty.")
django.core.exceptions.ImproperlyConfigured: The SECRET_KEY setting must not be empty.
```

I moved .env-dist to .env and added 'DJANGO_SECRET_KEY' to some random value but it looks like it is being blocked on connections to redis:

```
strace -fp 22522
strace: Process 22522 attached
connect(3, {sa_family=AF_INET, sin_port=htons(6379), sin_addr=inet_addr("104.239.213.7")}, 16) = -1 ETIMEDOUT (Connection timed out)
```

Any reason why collectstatic management command needs access to redis?